src/Graph/GraphPrinter.php

```diff
@@ -101,183 +112,203 @@ class GraphPrinter extends SMWResultPrinter {
+						}
 
-		if ( $this->m_graphLink ) {
-			$nodeLinkURL = "[[" . $text . "]]";
-		}
+						// assign color
+						$color = $this->graphColors[array_search( $parentNode['predicate'], $this->legendItem, true )];
 
-		$text = $this->getWordWrappedText( $text, $this->m_wordWrapLimit );
+						// show arrow label (graphLabel is misleading but kept for compatibility reasons)
+						if ( $this->showGraphLabel ) {
+							$graphInput .= "label=\"" . $parentNode['predicate'] . "\"";
+							if ( $this->showGraphColor ) {
+								$graphInput .= ",fontcolor=$color,";
+							}
+						}
 
-		if ( $this->m_nodeLabel === self::NODELABEL_DISPLAYTITLE && $object instanceof SMWWikiPageValue ) {
-			$objectDisplayTitle = $object->getDisplayTitle();
-			if ( !empty( $objectDisplayTitle )) {
-				$nodeLabel = $this->getWordWrappedText( $objectDisplayTitle, $this->m_wordWrapLimit );
+						// colorize arrow
+						if ( $this->showGraphColor ) {
+							$graphInput .= "color=$color";
+						}
+						$graphInput .= ']';
+					}
+				}
+				$graphInput .= ';';
 			}
 		}
+		$graphInput .= "}";
 
-		if ( $this->m_graphLink ) {
-			if( $nodeLabel === '' ) {
-				$graphInput .= " \"$text\" [URL = \"$nodeLinkURL\"]; ";
-			} else {
-				$graphInput .= " \"$text\" [URL = \"$nodeLinkURL\", label = \"$nodeLabel\"]; ";
-			}
-		}
 
-		if ( !$isName ) {
-			$graphInput .= $this->m_parentRelation ? " \"$text\" -> \"$name\" " : " \"$name\" -> \"$text\" ";
+		// Calls graphvizParserHook function from MediaWiki GraphViz extension
+		$result = $GLOBALS['wgParser']->recursiveTagParse( "<graphviz>$graphInput</graphviz>" );
+
 
-			if ( $this->m_graphLabel || $this->m_graphColor ) {
-				$graphInput .= ' [';
+		// append legend
+		if ( $this->showGraphLegend && $this->showGraphColor ) {
+			$itemsHtml = '';
+			$colorCount = 0;
+			$arraySize = count( $this->graphColors );
 
-				if ( array_search( $labelName, $this->m_labelArray, true ) === false ) {
-					$this->m_labelArray[] = $labelName;
+			foreach ( $this->legendItem as $legendLabel ) {
+				if ( $colorCount >= $arraySize ) {
+					$colorCount = 0;
 				}
 
-				$color = $this->m_graphColors[array_search( $labelName, $this->m_labelArray, true )];
+				$color = $this->graphColors[$colorCount];
+				$itemsHtml .= Html::rawElement( 'div', [ 'class' => 'graphlegenditem', 'style' => "color: $color" ],
+					"$color: $legendLabel" );
 
-				if ( $this->m_graphLabel ) {
-					$graphInput .= "label=\"$labelName\"";
-					if ( $this->m_graphColor ) {
-						$graphInput .= ",fontcolor=$color,";
-					}
-				}
+				$colorCount ++;
+			}
 
-				if ( $this->m_graphColor ) {
-					$graphInput .= "color=$color";
-				}
+			$result .= Html::rawElement( 'div', [ 'class' => 'graphlegend' ], "$itemsHtml" );
+		}
+		return $result;
+	}
 
-				$graphInput .= ']';
+	/**
+	 * Process a result row and create SRF\GraphNodes
+	 *
+	 * @since 2.5.0
+	 *
+	 * @param array $row
+	 * @param $outputmode
+	 * @param array $nodes
+	 *
+	 */
```

Comment:
Comment is wrong

Reply:
see: https://github.com/RacoonDev/SemanticResultFormats/commit/2733ba422c72be6b4d3d7c26720fac7b4fd27c3b